### PR TITLE
Structural changes to patterns subfolder

### LINKED ISSUE #114: Use categories for `patterns` subfolder
It could make sense to structure/sort some of the patterns into a subfolder named after corresponding categories as mentioned in `Design patterns (by Gamma, Helm, Johnson, Vlissides)`.

- `patterns/structural`
- `patterns/behavioural `
- `patterns/creational`

Patterns not fitting into any of the categories could stay in `patterns`. When we merged all the older PRs we could start with that process to avoid merge conflicts.

Thoughts?

### PR DESCRIPTION
**Pull request by simonsan**
I changed the subfolder structure to add patterns to their categories more easily.

Fixes #114

### CODE REVIEW

File: SUMMARY.md [hunk 1]
```diff
@@ -10,31 +10,33 @@
...
+    - [RAII Guards](./patterns/behavioural/RAII.md)
+    - [Strategy](./patterns/behavioural/strategy.md)
+    - [Visitor](./patterns/behavioural/visitor.md)
+  - [Creational](./patterns/creational/intro.md)
+    - [Builder](./patterns/creational/builder.md)
+    - [Fold](./patterns/creational/fold.md)
+  - [Structural](./patterns/structural/intro.md)
+    - [Compose Structs](./patterns/structural/compose-structs.md)
+    - [Prefer Small Crates](./patterns/structural/small-crates.md)
+    - [Contain unsafety in small modules](./patterns/structural/unsafe-mods.md)
```
Comment (simonsan) [inline SUMMARY.md:36]: @MarcoIeni Could you check if I sorted them right in here?

Comment (MarcoIeni) [reply SUMMARY.md:36]: you forgot Entry API

Comment (simonsan) [reply SUMMARY.md:36]: it's an empty template, we forgot this in there:
https://rust-unofficial.github.io/patterns/patterns/entry.html

Comment (MarcoIeni) [reply SUMMARY.md:36]: oh, sorry. The rest seems correct.

**Comment by simonsan** [comment]
I'll fix the conflicts

**Comment by simonsan** [comment]
@MarcoIeni Should be fixed

**Comment by simonsan** [comment]
I removed the `entry` and `late bounds` articles as they were basically empty.

File: patterns/behavioural/intro.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# Behavioural Patterns
+
+Design patterns that identify common communication patterns among objects.
+By doing so, these patterns increase flexibility in carrying out communication.
+[Wikipedia](https://en.wikipedia.org/wiki/Behavioral_pattern)
```
Comment (MarcoIeni) [inline patterns/behavioural/intro.md:5]:
```suggestion
From [Wikipedia](https://en.wikipedia.org/wiki/Behavioral_pattern):

> Design patterns that identify common communication patterns among objects.
> By doing so, these patterns increase flexibility in carrying out communication.
```

if you like this we could do it also for the other ones.

**Review by MarcoIeni — APPROVED** [review]

**Pull request merged**